src/lib/SaferNames/SourceRename.hs

```diff
@@ -116,15 +116,19 @@ instance SourceRenamableE (SourceNameOr UVar) where
       Just (EnvVal SuperclassNameRep _) -> error "Shouldn't find these in source map"
   sourceRenameE _ = error "Shouldn't be source-renaming internal names"
 
+lookupSourceName :: Renamer m => SourceName -> m n (EnvVal Name n)
+lookupSourceName v = do
+  SourceMap sourceMap <- askSourceMap
+  case M.lookup v sourceMap of
+    Nothing     -> throw UnboundVarErr $ pprint v
+    Just envVal -> return envVal
+
 instance NameColor c => SourceRenamableE (SourceNameOr (Name c)) where
   sourceRenameE (SourceName sourceName) = do
-    SourceMap sourceMap <- askSourceMap
-    case M.lookup sourceName sourceMap of
-      Nothing    -> throw UnboundVarErr $ pprint sourceName
-      Just (EnvVal rep val) ->
-        case eqNameColorRep rep (nameColorRep :: NameColorRep c) of
-          Just ColorsEqual -> return $ InternalName val
-          Nothing -> throw TypeErr $ "Incorrect name color: " ++ pprint sourceName
+    lookupSourceName sourceName >>= \case
+      EnvVal rep val -> case eqNameColorRep rep (nameColorRep :: NameColorRep c) of
+        Just ColorsEqual -> return $ InternalName val
+        Nothing -> throw TypeErr $ "Incorrect name color: " ++ pprint sourceName
```

Comment:
Should this error be a compiler error? The error message is pretty useless to the user, so if they ever see this I could consider it to be our fault.